include/grpc/event_engine/event_engine.h

```diff
@@ -271,36 +281,48 @@ class EventEngine {
     virtual void TryCancelLookup(LookupTaskHandle handle) = 0;
   };
 
+  /// At time of destruction, the EventEngine must have no active
+  /// responsibilities. EventEngine users (applications) are responsible for
+  /// cancelling all tasks and DNS lookups, shutting down listeners and
+  /// endpoints, prior to EventEngine destruction. If there are any outstanding
+  /// tasks, any running listeners, etc. at time of EventEngine destruction,
+  /// that is an invalid use of the API, and it will result in undefined
+  /// behavior.
   virtual ~EventEngine() = default;
 
   // TODO(nnoble): consider whether we can remove this method before we
   // de-experimentalize this API.
   virtual bool IsWorkerThread() = 0;
 
-  // TODO(hork): define return status codes
-  /// Retrieves an instance of a DNSResolver.
-  virtual absl::StatusOr<std::unique_ptr<DNSResolver>> GetDNSResolver() = 0;
+  /// Creates and returns an instance of a DNSResolver.
+  virtual std::unique_ptr<DNSResolver> GetDNSResolver() = 0;
 
-  /// Intended for future expansion of Task run functionality.
-  struct RunOptions {};
-  /// Run a callback as soon as possible.
+  /// Executes a callback as soon as possible.
   ///
   /// The \a fn callback's \a status argument is used to indicate whether it was
   /// executed normally. For example, the status may be CANCELLED if the
   /// EventEngine is being shut down. \a fn is guaranteed to be called exactly
   /// once.
-  virtual void Run(Callback fn, RunOptions opts) = 0;
+  ///
+  /// It is an error to call \a Run after the EventEngine is shut down.
+  /// The provided \a fn will be executed with an error status immediately,
+  /// possibly in the same call stack.
```

Comment:
Should these comments exist ? As per our recent discussions, no pending callbacks may exist after event engine is destroyed. Since there is no explicit shutdown method, I think these comments can be removed because the Run method cannot be invoked once the event engine is destroyed anyway.

Reply:
Good catch! That scenario is now impossible. I've deleted this paragraph.